nimare/base/estimators.py

```diff
@@ -24,26 +21,41 @@ def __init__(self):
     def transform(self, dataset):
         """Add stuff to transformer.
         """
-        if not isinstance(dataset, Dataset):
+        if not hasattr(dataset, 'slice'):
             raise ValueError('Argument "dataset" must be a valid Dataset '
                              'object, not a {0}'.format(type(dataset)))
 
 
 class Estimator(NiMAREBase):
     """Estimators take in Datasets and return MetaResults
     """
-    def __init__(self):
-        pass
 
-    @abstractmethod
-    def fit(self, dataset):
-        """Apply estimation to dataset and output results.
-        """
-        if not isinstance(dataset, Dataset):
+    # Inputs that must be available in input Dataset. Keys are names of
+    # attributes to set; values are strings indicating location in Dataset.
+    _inputs = {}
+
+    def _validate_input(self, dataset):
+        if not hasattr(dataset, 'slice'):
             raise ValueError('Argument "dataset" must be a valid Dataset '
                              'object, not a {0}'.format(type(dataset)))
+        for k, v in self._inputs.items():
+            data = dataset.get(v[0], **v[1])
+            if not data:
+                raise ValueError("Estimator {0} requires input dataset to "
+                                 "contain {1} with filters {2}, but none "
+                                 "were found.".format(self.__class__.__name__,
+                                                      v[0], v[1]))
+            setattr(self, k, data)
```

Comment:
I'm now thinking it might not the best practice to set attributes directly on the instance... we should probably rename `_inputs` to something more explicit, like `_required_inputs`, and then store the actual data in an `inputs` dictionary rather than setting them as attributes directly.

Reply:
I changed it a bit and added a really bad first attempt at a `Dataset.get` method, but it seems to be working for images and metadata at least.